Feature/blui 5674 header examples (#54)
etn-ccis/blui-react-native: docs/src/componentDocs/Header/markdown/HeaderExamples.mdx

import Box from '@mui/material/Box';
import headerImageExample from '../images/headerImage.png';
import headerBackgroundExample from '../images/headerBackgroundImage.png';
import headerActionsExample from '../images/headerActionsImage.png';
import headerSearchExample from '../images/headerSearchImage.png';
import headerCollapsedExample from '../images/headerCollapsedImage.png';
import headerExpandedExample from '../images/headerExpandedImage.png';

## Header

> The `<Header>` component is used at the top of a page to display information. It displays a title and can optionally show a subtitle, a background image, a navigation button, and multiple action buttons. The header can also be configured to expand or collapse as needed. For expand/collapse behavior tied to the screen scroll position, you should use the [`<CollapsibleHeaderLayout>`](/components/collapsible-header-layout/examples) component.

Reviewer: ```suggestion
A basic `Header` component is used at the top of the screen and can show a title and (optional) subtitle. A third line of text can also be configured using the `info` prop, but it will only be visible when the `Header` is expanded (see below). 
```

<Box
  sx={{
    width: '100%',
    textAlign: 'center',
  }}>
  <Box
    sx={{
      width: '100%',
      maxWidth: '500px',
    }}
    component="img"
    alt="Header"
    src={headerImageExample}
  />
</Box>

```
<Header
    title={'Valley Forge'}
    subtitle={'The Last Stand'}
/>;
```

> ## Header with Background image

Reviewer: ```suggestion
## Changing the Background

You can apply a simple background to the Header using the `backgroundColor` prop. You can also apply an image background using `backgroundImage`. If both are applied, they will be blended together.
```

<Box
  sx={{
    width: '100%',
    textAlign: 'center',
  }}>
  <Box
    sx={{
      width: '100%',
      maxWidth: '500px',
    }}
    component="img"
    alt="Header background image"
    src={headerBackgroundExample}
  />
</Box>

```
<Header
    title={'Valley Forge'}
    subtitle={'The Last Stand'}
    backgroundImage={farm.jpg}
/>;
```

> ## Header with Actions and Menu

Reviewer: ```suggestion
## Adding Actions

You can add custom icon buttons to the top right corner of the Header. You can also configure an icon in the top left for typical navigation functions (such as a menu icon or a back icon, depending on the context).
```

<Box
  sx={{
    width: '100%',
    textAlign: 'center',
  }}>
  <Box
    sx={{
      width: '100%',
      maxWidth: '500px',
    }}
    component="img"
    alt="Header actions"
    src={headerActionsExample}
  />
</Box>

```
<Header
    title={'Valley Forge'}
    subtitle={'The Last Stand'}
    actionItems={[
        {
            icon: { name: 'settings'},
            onPress: () => {},
        },
        {
            icon: { name: 'more-vert' },
            onPress: () => {},
        },
    ]}
    icon={{ name: 'menu' }}
/>;
```

> ## Header with Search

Reviewer: ```suggestion
## Adding a Search Bar

The `Header` includes support for a built-in search bar. If configured, the search button will be appended to any provided action items.
```

<Box
  sx={{
    width: '100%',
    textAlign: 'center',
  }}>
  <Box
    sx={{
      width: '100%',
      maxWidth: '500px',
    }}
    component="img"
    alt="Header search"
    src={headerSearchExample}
  />
</Box>

```
<Header
    title={'Valley Forge'}
    subtitle={'The Last Stand'}
    actionItems={[
        {
            icon: { name: 'settings'},
            onPress: () => {},
        },
        {
            icon: { name: 'more-vert' },
            onPress: () => {},
        },
    ]}
    icon={{ name: 'menu' }}
    searchableConfig={{ onChangeText: () => {} }}
/>;
```

> ## Header with Collapsing and Expanding

Reviewer: ```suggestion
## Expanding and Collapsing

The `Header` can be configured to expand and collapse from a large size to a small size. If you want the `Header` to expand and collapse automatically as the page is scrolled, check out the [`CollapsibleHeaderLayout`](/components/collapsible-header-layout/examples) component.
```

<Box
  sx={{
    display: 'flex',
    justifyContent: 'space-evenly',
    alignItems: 'center',
    mb: 2,
    width: '100%',
    textAlign: 'center',
  }}>
  <Box
    sx={{
      width: '40%',
    }}
    component="img"
    alt="Header collapsed"
    src={headerCollapsedExample}
  />
  <Box
    sx={{
      width: '40%',
    }}
    component="img"
    alt="Header expanded"
    src={headerExpandedExample}
  />
</Box>

```
<Header
    title={'Valley Forge'}
    subtitle={'The Last Stand'}
    info={'Information'}
    backgroundImage={farm.jpg}
    icon={{ name: 'menu' }}
    onIconPress={() => {}}
    actionItems={[
        {
            icon: { name: 'settings'},
            onPress: () => {},
        },
        {
            icon: { name: 'more-vert' },
            onPress: () => {},
        },
    ]}
    variant="static"
    icon={{ name: 'menu' }}

>     searchableConfig={{ onChangeText: () => {} }}

Reviewer: You can leave off all of this stuff that is not related to the expand/collapse behavior we are demonstrating in this example.
